lambda_code/src/fhir_repository.py

```diff
@@ -67,33 +63,30 @@ def __init__(self, imms: dict, patient: dict):
         """Create attributes that may be used in dynamodb table"""
         imms_id = imms["id"]
         self.pk = _make_immunization_pk(imms_id)
-        self.patient_pk = _make_patient_pk(
-            [x for x in imms["contained"] if x.get("resourceType") == "Patient"][0][
-                "identifier"
-            ][0]["value"]
-        )
+        if patient:
```

Comment:
NHS number may not exist, even if patient does. Does this scenario need to be caught?

Reply:
No, this is patient we get back from PDS. If we don't have a NHS Number, we're not calling PDS, so patient is set to `{}` (which is false when tested).